test/integration/client/client_test.go

```diff
@@ -1342,3 +1350,112 @@ func TestExtractModifyApply_ForceOwnership(t *testing.T) {
 		t.Errorf("createMgrExtracted apply configuration did not match expected, got:\n%s\n", cmp.Diff(expectedCreateExtracted, createMgrExtracted))
 	}
 }
+
+// regression test for https://issues.k8s.io/117258
+func TestAPIServerTransportMetrics(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, "AllAlpha", true)()
+	defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, "AllBeta", true)()
+
+	result := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{"--disable-admission-plugins", "ServiceAccount"}, framework.SharedEtcd())
+	defer result.TearDownFn()
+
+	client := clientset.NewForConfigOrDie(result.ClientConfig)
+
+	hits1, misses1, entries1 := checkTransportMetrics(t, client)
+	// hit ratio at startup depends on multiple factors
+	// be conservative and expect only a 75 percent
+	if (hits1*100)/(hits1+misses1) < 75 {
```

Comment:
expecting repeated construction during startup seems weird... wouldn't an ideal startup construct clients exactly once, and get a hit rate of 0?

where is the hit rate currently?

Reply:
there are multiple process building clients, and some of them like the one this covers the regression were not cacheable ... also this runs with a default configuration, I can imaging that enabling new features may enable new controllers that build their own clientsets

The rate is ~ 88 IIRC

Reply:
I'm wondering if resetting metrics at the start of the test would improve what we see here

Reply:
I grepped around and it seems there is no reset for metrics on the integration tests
https://github.com/kubernetes/kubernetes/blob/4cff1311c5543af30a3cbb3fa873b3fadf9ccf7a/test/integration/metrics/metrics_test.go

but I would add it so we always reset the global registry at the start of the test

Reply:
awesome 
```
	// reset default registry metrics
	legacyregistry.Reset()
```
actually resets all the metrics